source/opencascade/utilities.cc

```diff
@@ -267,6 +275,66 @@ namespace OpenCASCADE
     AssertThrow(OK, ExcMessage("Failed to write IGES file."));
   }
 
+
+  TopoDS_Shape
+  read_STL(const std::string &filename)
+  {
+    StlAPI_Reader reader;
+    TopoDS_Shape  shape;
+    reader.Read(shape, filename.c_str());
+    return shape;
+  }
+
+  void
+  write_STL(const TopoDS_Shape &shape,
+            const std::string & filename,
+            const double        deflection,
+            const bool          sew_different_faces,
+            const double        sewer_tolerance,
+            const bool          is_relative,
+            const double        angular_deflection,
+            const bool          in_parallel)
+  {
+    TopLoc_Location            Loc;
+    std::vector<TopoDS_Vertex> vertices;
+    std::vector<TopoDS_Edge>   edges;
+    std::vector<TopoDS_Face>   faces;
+    OpenCASCADE::extract_geometrical_shapes(shape, faces, edges, vertices);
+    bool mesh_is_present = true;
+    for (unsigned int i = 0; i < faces.size(); ++i)
+      {
+        Handle(Poly_Triangulation) theTriangulation =
+          BRep_Tool::Triangulation(faces[i], Loc);
+        if (theTriangulation.IsNull())
+          mesh_is_present = false;
```

Comment:
can you add a break after this? You don't need to check anymore faces if there is at least one face which is not meshed.

Reply:
done